File: inno-erp/inno_erp/inno_setup/overrides/brand/brand.py
from erpnext.setup.doctype.brand.brand import Brand
from frappe.utils import get_abbr

from inno_erp.utils.string import scrub


class InnoBrand(Brand):
	def validate(self):
		if not self.custom_abbr:
			self.custom_abbr = get_abbr(self.name, 3) if " " in self.name else self.name

		self.custom_abbr = scrub(self.custom_abbr, separator="-")

File: inno-erp/inno_erp/inno_setup/overrides/brand/brand.js
frappe.ui.form.on("Brand", {
	brand: (frm) => {
		if (frm.doc.brand.includes(" ")) {
			frm.set_value("custom_abbr", frappe.get_abbr(frm.doc.brand));
			return;
		}
		frm.set_value("custom_abbr", frm.doc.brand);
	},
});
